### PR TITLE
Issue #4119: fixed verifying tokens being referenced in configs

### PR DESCRIPTION
**Pull request by rnveach**
Issue #4119

No new issues created yet.
Do we need issues for our config?
I will update CS 8 wiki for these google issues

### CODE REVIEW

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 1]
```diff
@@ -130,7 +130,8 @@
         // https://github.com/checkstyle/checkstyle/issues/3730
         CHECKSTYLE_TOKENS_IN_CONFIG_TO_IGNORE.put("AnnotationLocation",
                 Stream.of("TYPECAST", "DOT", "TYPE_ARGUMENT", "LITERAL_NEW", "LITERAL_THROWS",
-                        "IMPLEMENTS_CLAUSE").collect(Collectors.toSet()));
+                        "IMPLEMENTS_CLAUSE", "CLASS_DEF", "CTOR_DEF", "ENUM_DEF", "INTERFACE_DEF",
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: No new issue needed, will update existing one #3730.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 2]
```diff
@@ -190,8 +203,10 @@
                 // can be empty for special cases via '6.2 Caught exceptions: not ignored'
                 "LITERAL_CATCH",
                 // specifically allowed via '5.2.4 Constant names'
-                "ARRAY_INIT")
-                .collect(Collectors.toSet()));
+                "ARRAY_INIT",
+                // state of the configuration when test was made until
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: new issue at #4121

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 3]
```diff
@@ -208,7 +223,9 @@
                 // assignment operators and they are allowed to break before or after the symbol
                 "DIV_ASSIGN", "BOR_ASSIGN", "SL_ASSIGN", "ASSIGN", "BSR_ASSIGN", "BAND_ASSIGN",
                 "PLUS_ASSIGN", "MINUS_ASSIGN", "SR_ASSIGN", "STAR_ASSIGN", "BXOR_ASSIGN",
-                "MOD_ASSIGN").collect(Collectors.toSet()));
+                "MOD_ASSIGN",
+                // state of the configuration when test was made until
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: new issue at #4122

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 4]
```diff
@@ -158,6 +161,16 @@
...
                 "WILDCARD_TYPE", "GENERIC_END", "GENERIC_START").collect(Collectors.toSet()));
+        CHECKSTYLE_TOKENS_IN_CONFIG_TO_IGNORE.put("EmptyBlock", Stream.of(
+                // state of the configuration when test was made until
+                "INSTANCE_INIT", "LITERAL_DO", "LITERAL_ELSE", "LITERAL_FINALLY", "LITERAL_FOR",
+                "LITERAL_IF", "LITERAL_SWITCH", "LITERAL_SYNCHRONIZED", "LITERAL_TRY",
+                "LITERAL_WHILE", "STATIC_INIT")
+                .collect(Collectors.toSet()));
+        CHECKSTYLE_TOKENS_IN_CONFIG_TO_IGNORE.put("RightCurly", Stream.of(
+                // state of the configuration when test was made until
+                "LITERAL_CATCH", "LITERAL_ELSE", "LITERAL_FINALLY", "LITERAL_IF", "LITERAL_TRY")
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: I think we should enable all. It looks like all should be `alone`.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 5]
```diff
@@ -158,6 +161,16 @@
                 "ARRAY_INIT",
                 // these are covered by GenericWhitespaceCheck
                 "WILDCARD_TYPE", "GENERIC_END", "GENERIC_START").collect(Collectors.toSet()));
+        CHECKSTYLE_TOKENS_IN_CONFIG_TO_IGNORE.put("EmptyBlock", Stream.of(
+                // state of the configuration when test was made until
+                "INSTANCE_INIT", "LITERAL_DO", "LITERAL_ELSE", "LITERAL_FINALLY", "LITERAL_FOR",
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: I think we should enable all.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java [hunk 6]
```diff
@@ -141,7 +142,9 @@
                 .collect(Collectors.toSet()));
         CHECKSTYLE_TOKENS_IN_CONFIG_TO_IGNORE.put("NoWhitespaceAfter", Stream.of(
                 // whitespace after is preferred
-                "TYPECAST", "LITERAL_SYNCHRONIZED").collect(Collectors.toSet()));
+                "TYPECAST", "LITERAL_SYNCHRONIZED",
+                // state of the configuration when test was made until
+                "INDEX_OP").collect(Collectors.toSet()));
```
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: http://checkstyle.sourceforge.net/apidocs/com/puppycrawl/tools/checkstyle/api/TokenTypes.html#INDEX_OP
I don't see a reason why we shouldn't enable it.

**Comment by rnveach** [comment]
@romani Please confirm if you are ok with described changes to our config. All google problems have been made into new issues.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 5 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: agree

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 6 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: agree

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 1 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: agree

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 4 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: default of RightCurly is `same`.
I think we should be same https://github.com/checkstyle/checkstyle/blob/master/config/checkstyle_checks.xml#L185

Am I miss smth ?

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 2 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: agree

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 3 above)
Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: agree

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 1 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 2 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 6 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 5 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: Done.

File: src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java (hunk 4 above)
Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: > default of RightCurly is same.
I think we should be same 
Am I miss smth ?

Let's take `LITERAL_IF` and `LITERAL_ELSE` as an example from the ones missing.
If we switched them to `same` then we would have to write if/else like this:
````
    if (condition) {
       ...
    } else {
       ...
    }
````
while, as of right now, most of our if/else is written as:
````
    if (condition) {
       ...
    }
    else {
       ...
    }
````
Violations for `alone` are 9. Violations for `same` are 904.
We like writting our curlies alone on the line.

Comment (romani) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: YES, we need to make them to follow `alone`.

Comment (rnveach) [inline src/test/java/com/puppycrawl/tools/checkstyle/internal/AllChecksTest.java]: Done.

**Comment by rnveach** [comment]
@romani All done.

**Pull request merged**